engineio/src/transports/websocket.rs

```diff
@@ -1,122 +1,67 @@
-use crate::error::{Error, Result};
-use crate::packet::Packet;
-use crate::packet::PacketId;
-use crate::transport::Transport;
-use bytes::{BufMut, Bytes, BytesMut};
-use std::borrow::Cow;
-use std::str::from_utf8;
-use std::sync::{Arc, Mutex, RwLock};
-use websocket::{
-    client::Url, dataframe::Opcode, header::Headers, receiver::Reader, sync::stream::TcpStream,
-    sync::Writer, ws::dataframe::DataFrame, ClientBuilder as WsClientBuilder, Message,
-};
+use crate::error::Result;
+use crate::header::HeaderMap;
+use crate::transport::{AsyncTransport, Transport};
+use crate::transports_async::websocket::AsyncWebsocketTransport;
+use bytes::Bytes;
+use std::sync::Arc;
+use tokio::runtime::Runtime;
+use url::Url;
 
 #[derive(Clone)]
 pub struct WebsocketTransport {
-    sender: Arc<Mutex<Writer<TcpStream>>>,
-    receiver: Arc<Mutex<Reader<TcpStream>>>,
-    base_url: Arc<RwLock<url::Url>>,
+    runtime: Arc<Runtime>,
+    inner: Arc<AsyncWebsocketTransport>,
 }
```

Comment:
No need to do this in this PR but this struct looks more like a generic wrapper around an AsyncTransport implementing Transport

Reply:
Yes it actually is. I am about to build an async version and I think this will be subject to change soon!